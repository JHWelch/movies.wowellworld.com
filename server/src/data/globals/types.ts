import { MovieDto } from '@shared/dtos'
import { Timestamp } from 'firebase/firestore'

export type LastUpdated = {
  updatedEvents: number
  previousLastUpdated: Timestamp | null
  newLastUpdated: Timestamp | null
  tmdbMoviesSynced: MovieDto[]
}
