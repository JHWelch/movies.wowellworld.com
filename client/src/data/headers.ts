export const jsonHeaders = {
  'Content-Type': 'application/json',
  Accept: 'application/json',
}
