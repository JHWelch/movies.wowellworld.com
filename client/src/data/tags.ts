export const tags: {
  [key: string]: {
    query: string
    display: string
    description?: string
  }
} = {
  october: {
    query: 'Horror October',
    display: 'Horror October',
    description: 'What’s better than a scary movie? A scary movie with friends, of course! Join us throughout October if you dare for special screenings of horror and genre films every night. Get ready for the spooky, the creepy, the freaky, the fantastical, the grotesque, and so much more. Most movies start at 6, some start at other times so check the website! Please RSVP via the site or text so we know you’re coming, but please do come!',
  },
}
