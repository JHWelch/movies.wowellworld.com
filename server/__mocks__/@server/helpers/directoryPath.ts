import { jest } from '@jest/globals'

const directoryPath = jest.fn(() => '/path/to/dir')

export default directoryPath
