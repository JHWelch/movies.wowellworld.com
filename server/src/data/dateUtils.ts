export function today (): string {
  return dateToString(new Date())
}

export function tomorrow (): string {
  const tomorrow = new Date()
  tomorrow.setDate(tomorrow.getDate() + 1)

  return dateToString(tomorrow)
}

export function dateToString (date: Date): string {
  const array = date.toLocaleString('en-US', {
    timeZone: 'America/Chicago',
    year: 'numeric',
    month: '2-digit',
    day: '2-digit' ,
  }).split('/')

  return [
    array[2],
    array[0],
    array[1],
  ].join('-')
}
