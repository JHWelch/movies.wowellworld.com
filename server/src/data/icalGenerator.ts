
import { Week } from '@server/models/week'
import * as ics from 'ics'

export const icalGenerator = (week: Week): string | undefined => {
  const { error, value } = ics.createEvent({
    uid: week.dateString + '@movies.wowellworld.com',
    start: week.date.getTime(),
    duration: { minutes: week.totalLength },
    title: week.theme,
    description: 'Come one, come all (okay maybe not all) to the Welch-Cowell/Cowell-Welch/Wowell household for a recurring Thursday Night Movie Club!',
    location: '2111 W Argyle St. Chicago, IL 60625',
    url: 'https://movies.wowellworld.com/',
    status: 'CONFIRMED',
    busyStatus: 'BUSY',
    organizer: { name: 'Rachel & Jordan', email: 'fam@wowellworld.com' },
  })

  if (error) {
    console.error(error)

    return undefined
  }

  return value
}
