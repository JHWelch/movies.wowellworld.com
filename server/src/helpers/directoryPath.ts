import path from 'path'
import { fileURLToPath } from 'url'

const directoryPath = () => path.dirname(fileURLToPath(import.meta.url))

export default directoryPath
