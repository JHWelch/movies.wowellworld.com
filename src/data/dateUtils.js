export default class DateUtils {
  static getThursdayDate() {
    const today = new Date();
    const thursday = new Date(today.setDate(today.getDate() - today.getDay() + 4));

    return thursday;
  }

  static getThursday() {
    return this.getThursdayDate().toISOString().substring(0, 10);
  }

  static getNextTwoThursdays() {
    const thursday = this.getThursdayDate();

    return [
      new Date(thursday.setDate(thursday.getDate() + 7)).toISOString().substring(0, 10),
      new Date(thursday.setDate(thursday.getDate() + 7)).toISOString().substring(0, 10),
    ];
  }
}
