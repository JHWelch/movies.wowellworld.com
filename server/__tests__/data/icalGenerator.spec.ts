import { beforeEach, expect, it, jest } from '@jest/globals'
import { icalGenerator } from '@server/data/icalGenerator'
import directoryPath from '@server/helpers/directoryPath'
import MovieFactory from '@tests/support/factories/movieFactory'
import WeekFactory from '@tests/support/factories/weekFactory'
import MockDate from 'mockdate'

beforeEach(() => {
  MockDate.set('2021-01-01');
  (directoryPath as unknown as jest.Mock)
    .mockReturnValue(__dirname + '/../../src/data')
})

it('can generate an ical file from a week', async () => {
  const week = new WeekFactory().make()
  week.movies = [
    new MovieFactory().make({
      title: 'Labyrinth',
      director: 'Jim Henson',
      year:1986,
      url: 'https://example.com/labyrinth',
      tmdbId: 1234,
      posterPath: '/path/to/labyrinth.jpg',
      notionId: 'notionId1',
      theaterName: 'Theater',
      showingUrl: 'https://cal.example.com/labyrinth',
      time: '6:00 PM',
      length: 105,
    }),
    new MovieFactory().make({
      title: 'The NeverEnding Story',
      director: 'Wolfgang Petersen',
      year: 1984,
      url: 'https://example.com/neverending-story',
      tmdbId: 1234,
      posterPath: '/path/to/neverending_story.jpg',
      notionId: 'notionId2',
      theaterName: 'Theater',
      showingUrl: 'https://cal.example.com/neverending-story',
      time: '8:00 PM',
    }),
  ]

  expect(await icalGenerator(week)).toMatchSnapshot()
})
