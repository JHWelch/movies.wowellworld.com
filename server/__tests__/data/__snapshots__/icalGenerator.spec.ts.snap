// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can generate an ical file from an event 1`] = `
"BEGIN:VCALENDAR
VERSION:2.0
CALSCALE:GREGORIAN
PRODID:adamgibbons/ics
METHOD:PUBLISH
X-PUBLISHED-TTL:PT1H
BEGIN:VEVENT
UID:2021-09-13@movies.wowellworld.com
SUMMARY:Movie Nights: theme
DTSTAMP:20210101T000000Z
DTSTART:20210913T173000Z
DESCRIPTION:Come one\\, come all (okay maybe not all) to the Welch-Cowell/Co
	well-Welch/Wowell household for a recurring thursday Movie Night!\\n\\nThis 
	week's theme is theme.\\n\\n6:00 PM - Labyrinth (1986)\\n8:00 PM - The NeverE
	nding Story (1984)
URL:https://movies.wowellworld.com/
LOCATION:2111 W Argyle St. Chicago\\, IL 60625
STATUS:CONFIRMED
ORGANIZER;CN="Rachel & Jordan":MAILTO:fam@wowellworld.com
X-MICROSOFT-CDO-BUSYSTATUS:BUSY
X-ALT-DESC;FMTTYPE=text/html:<p><html><body>   <p>Come one, come all (okay 
	maybe not all) to the Welch-Cowell/Cowell-Welch/Wowell household for a rec
	urring thursday Movie Night!</p>    <p>This week\\'s theme is <b>theme</b>.
	</p>    <ul>            <li>         <b>6:00 PM</b>         -         <i>L
	abyrinth</i>         (1986)       </li>            <li>         <b>8:00 PM
	</b>         -         <i>The NeverEnding Story</i>         (1984)       <
	/li>        </ul> </body></html> </p>
DURATION:PT240M
END:VEVENT
END:VCALENDAR
"
`;
